src/Nethermind/Nethermind.Evm/Precompiles/Secp256r1Precompile.cs

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Security.Cryptography;
+using Nethermind.Core;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+
+namespace Nethermind.Evm.Precompiles;
+
+public class Secp256r1Precompile : IPrecompile<Secp256r1Precompile>
+{
+    private static readonly byte[] ValidResult = new byte[] { 1 }.PadLeft(32);
+
+    public static readonly Secp256r1Precompile Instance = new();
+    public static Address Address { get; } = Address.FromNumber(0x100);
+
+    public long BaseGasCost(IReleaseSpec releaseSpec) => 3450L;
+    public long DataGasCost(in ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec) => 0L;
+
+    // TODO can be optimized - Go implementation is 2-6 times faster depending on the platform. Options:
+    // - Try to replicate Go version in C#
+    // - Compile Go code into a library and call it via P/Invoke
+    public (ReadOnlyMemory<byte>, bool) Run(in ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec)
+    {
+        if (inputData.Length != 160)
+            return (null, true);
```

Comment:
should it be `(null, true)` or `IPrecompile.Failure`?

Reply:
They decided to return always true (and did not spec it well) and now do not want to change: https://ethereum-magicians.org/t/eip-7212-precompiled-for-secp256r1-curve-support/14789/104